.github/workflows/build-firmware.yaml

```diff
@@ -3,8 +3,54 @@ name: Firmware CI
 on: [push,pull_request]
 
 jobs:
-  build:
+  config:
     runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      # We can tell if this is a push rather than a pull request by testing for the existence of the 'pusher' object
+      # We have to do this on each step in this job rather than the whole job, or everything that needs this job will be skipped.
+      if: ${{ github.event.pusher }}
+      # Because we are making changes, we need to fetch all history
+      with:
+        fetch-depth: 0
+
+    - name: Generate Configs
+      if: ${{ github.event.pusher }}
+      working-directory: ./firmware/
+      run: sh gen_config.sh
+
+    - name: Generate Live Documentation
+      if: ${{ github.event.pusher }}
+      working-directory: ./firmware/
+      run: sh gen_live_documentation.sh
+
+    # Commit changes and catch the error that occurs if nothing has been changed (without catching other errors)
+    - name: Commit changes
+      if: ${{ github.event.pusher }}
+      run: |
+        echo ${{ github.event.pusher }}
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub build-firmware Action"
+        git commit -am "Auto-generated configs and docs" 2>&1 | grep -E '(nothing to commit|changed)'
```

Comment:
we cannot have builds that commit over the top of your pull request - this sort of thing should only run on push to master

Reply:
This will only run if it's a push.

Reply:
Yes - it will commit over the top of your PR branch when you push to it.  We need to restrict commits to only the rusefi/master branch, not mck1117/whatever-feature-27

Reply:
Overzealous auto-commits are how you pollute the commit history with a thousand merge commits when the action has added new commits while you were still working locally

Reply:
So only commit if the push is to master?

Reply:
Yep!  Though we have to check that it's the "real" master, and not a fork's master.